utils.lua

```diff
@@ -162,4 +162,27 @@ function nn.utils.contiguousView(output, input, ...)
   return output
 end
 
+-- go over specified fields and clear them. accepts
+-- nn.utils.clearState(self, {'_buffer', '_buffer2'}) and
+-- nn.utils.clearState(self, '_buffer', '_buffer2')
+function nn.utils.clear(self, ...)
+   local arg = {...}
+   if #arg > 0 and type(arg[1]) == 'table' then
+      arg = arg[1]
+   end
+   local function clear(f)
+      if self[f] then
+         if torch.isTensor(self[f]) then
+            self[f]:set()
+         elseif type(self[f]) == 'table' then
+            self[f] = {}
```

Comment:
What do you think about calling `collectgarbage()` at he end of `nn.utils.clear()` to free tensors that are no longer referenced (e.g. stuff built by [zeroTableCopy()](https://github.com/torch/nn/blob/0c77df0fcae2fdb0f2902fc88a57d30f3fa75cc2/SelectTable.lua#L20) in SelectTable) after the `= {}` replacement?